src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```diff
@@ -0,0 +1,299 @@
+import { Component, ElementRef, HostListener, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { AttachmentService } from 'app/lecture/attachment.service';
+import * as PDFJS from 'pdfjs-dist';
+import 'pdfjs-dist/build/pdf.worker';
+import { Attachment } from 'app/entities/attachment.model';
+import { AttachmentUnit } from 'app/entities/lecture-unit/attachmentUnit.model';
+import { AttachmentUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/attachmentUnit.service';
+import { onError } from 'app/shared/util/global.utils';
+import { AlertService } from 'app/core/util/alert.service';
+import { Subscription } from 'rxjs';
+import { Course } from 'app/entities/course.model';
+import { HttpErrorResponse } from '@angular/common/http';
+
+type NavigationDirection = 'next' | 'prev';
+
+@Component({
+    selector: 'jhi-pdf-preview-component',
+    templateUrl: './pdf-preview.component.html',
+    styleUrls: ['./pdf-preview.component.scss'],
+})
+export class PdfPreviewComponent implements OnInit, OnDestroy {
+    @ViewChild('pdfContainer', { static: true }) pdfContainer: ElementRef<HTMLDivElement>;
+    @ViewChild('enlargedCanvas') enlargedCanvas: ElementRef<HTMLCanvasElement>;
+
+    readonly DEFAULT_SLIDE_WIDTH = 250;
+    course?: Course;
+    attachment?: Attachment;
+    attachmentUnit?: AttachmentUnit;
+    isEnlargedView = false;
+    currentPage = 1;
+    totalPages = 0;
+    attachmentSub: Subscription;
+    attachmentUnitSub: Subscription;
+
+    constructor(
+        public route: ActivatedRoute,
+        private attachmentService: AttachmentService,
+        private attachmentUnitService: AttachmentUnitService,
+        private alertService: AlertService,
+    ) {}
+
+    ngOnInit() {
+        this.route.data.subscribe((data) => {
+            this.course = data.course;
+            if ('attachment' in data) {
+                this.attachment = data.attachment;
+                this.attachmentSub = this.attachmentService.getAttachmentFile(this.course?.id!, this.attachment?.id!).subscribe({
+                    next: (blob: Blob) => this.loadPdf(URL.createObjectURL(blob)),
+                    error: (error: HttpErrorResponse) => onError(this.alertService, error),
+                });
+            } else if ('attachmentUnit' in data) {
+                this.attachmentUnit = data.attachmentUnit;
+                this.attachmentUnitSub = this.attachmentUnitService.getAttachmentFile(this.course?.id!, this.attachmentUnit?.id!).subscribe({
```

Comment:
If you're confident that `this.course.id` is not `undefined`, you should use `this.course!.id!` as otherwise, you're telling typescript that the value can and cannot be `undefined` at the same time. Same for `attachmentUnit`

```suggestion
                this.attachmentUnitSub = this.attachmentUnitService.getAttachmentFile(this.course!.id!, this.attachmentUnit!.id!).subscribe({
```

Reply:
Fixed